### DISCUSSION TITLE
Add PiperTTS to the WebUI?

### DISCUSSION THREAD

**Discussion by FemBoxbrawl** [rsxdalv/TTS-WebUI#214] [Ideas]
I watched a video of Natlamir who made like a UI for PiperTTS, which is a super fast, AI TTS generation, It's really really fast. The only problem is I my Antivirus discovered one file  with in it as a virus.. so I do not know. but maybe the original piperTTS would work.

I would love to have a super fast generation and was wondering if you could maybe add it?

ofcourse you do not have to.

**Comment by rsxdalv** [comment]
While I can't recommend not using an antivirus, I feel that many things
could cause it to give a false reading. Actually, if you have reason to
think it's a valid file without a virus, you can report it to the company.

Before I do that, make sure that my UI doesn't upset your antivirus.

I think I heard of PiperTTS before, I will take a look at it next month.

**Reply by LykosTactical** [reply]
Hello there! :) Can we PLEASE revisit adding PiperTTS into this project? Its exceptionally easy to build for both ARM64 and AMD64, it is very, very fast with pure CPU, and all it would need is webUI integration into your project. The voice output quality is very good as well. 

Devs have integrated Piper into the project "LocalAI" to generate voice, however the LocalAI project's implementation of Piper is simplistic to say the least, and LocalAI's audio generation pales in comparison to your tts-generation-webui project and features. I feel like adding Piper into your project would be an incredibly good fit and would help support even more systems and faster voice generation when needed.

Worst case, I'm considering making Grok3 build it all in for me and try to add it into your project, but thats typically a code and build iteration nightmare for me that takes a long time to get it done manually if I had to do it myself with Grok or OpenAI, versus just having the expert Dev that built this project go and add it themselves. I can promise you that the Piper build is very easy though.

I hope you'll consider. Thank you in advance and also a BIG THANK YOU for building this project to begin with! It's excellent and top-notch to have all of this locally. I hope this message finds you well.

**Reply by rsxdalv** [reply]
Here's the issue where it's been discussed:
https://github.com/rsxdalv/tts-generation-webui/issues/191

I think I have spent 4-8 hours trying to make piper 'nice' and installable but didn't get too far. Though I have developed with C++, building Piper TTS into a wheel has been a major pain. I might try again at some point, but the fact that it has been easier to do compile much bigger projects, such as deepspeed and xformers makes me feel a bit cold towards Piper. But maybe now I have the experience that will help me actually get piper pip-wheels.
If we use an externally installed piper, then it's trivial for me to just call the command and get the output. However, most extensions aim to be 'clean installs'.

**Reply by rsxdalv** [reply]
Thanks - building the .exe is easy indeed; however, once I tried to run it on Windows it managed to crash my WSL.
When I tried to run it via terminal, it only works with --help, anything else does not output anything. But I found the issue, the .onnx is mandatory.
So now I'm facing a bit of a problem - distribution. I can have users download it themselves and make an extension that uses it. Downloading the 'exe' with a script will easily make every antivirus angry.
Building it directly as a wheel seems to be still quite challenging, but I will give it a try.

**Reply by rsxdalv** [reply]
Side note - I also have a GPT SoVITS extension which is related to VITS and VITS is related to Piper TTS.

**Reply by rsxdalv** [reply]
Well, the impossible has happened I guess, Piper TTS has been added.
![image](https://github.com/user-attachments/assets/9661caa8-e3b6-4cd4-ab23-8f57ccd9bfb3)
